### PR TITLE
Add missing parameters to ModelCheckpoint CallbackParams

### PR DESCRIPTION
**Pull request by markelsanz14**
# What does this PR do ?
Adds missing parameters to ModelCheckpoint CallbackParams, to achieve feature parity with PTL. See ModelCheckpoint API here: https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.ModelCheckpoint.html

**Collection**: exp_manager

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -96,7 +96,10 @@ class CallbackParams:
     save_top_k: Optional[int] = 3
     save_weights_only: Optional[bool] = False
     mode: Optional[str] = "min"
+    auto_insert_metric_name: bool = True
```
Comment (MaximumEntropy) [inline nemo/utils/exp_manager.py:99]: What does this bit do? And how does it interact with specifying a filename that already has metrics specified like this in `megatron_gpt_config.yaml`. 

Ex: `filename: 'megatron_gpt--{val_loss:.2f}-{step}-{consumed_samples}`

Comment (markelsanz14) [reply nemo/utils/exp_manager.py:99]: When this is true, it will not break anything and behaves the same as until now. It's already true by default on PTL. When it's true, PTL will replace {val_loss:.2f} with something like val_loss=0.12-.
The equal (=) sign and the dash (-) signs are what PTL auto-inserts to the metric name. So it won't change anything.

However, when set to False, you can specify the name of the checkpoint manually, to replace some of those characters with any character you want. This can help avoid logging issues with W&B, TB, or other systems that don't like the = and - characters, so it gives you full control over the checkpoint name.

**Comment by nithinraok** [comment]
LGTM. `every_n_train_steps` is crucial as we move our training strategies to steps based rather than epoch based

**Review by nithinraok — DISMISSED** [review]

File: nemo/utils/exp_manager.py [hunk 2]
```diff
@@ -96,7 +96,10 @@ class CallbackParams:
...
     every_n_epochs: Optional[int] = 1
+    every_n_train_steps: Optional[int] = None
+    train_time_interval: Optional[timedelta] = None
```
Comment (titu1994) [inline nemo/utils/exp_manager.py:102]: Looks good. Maybe we should look into some way of testing the callbacks data class (if it exists in PTL) against exp managers checkpoint subconfig similar to how we check PTL trainer args against our trainer config.

**Review by titu1994 — APPROVED** [review]

File: nemo/utils/exp_manager.py (hunk 2 above)
Comment (markelsanz14) [reply nemo/utils/exp_manager.py:102]: It doesn't exist as far as I know, I think it's just the parameters of the ModelCheckpoint init function. But I'm not 100% sure, I haven't looked at it in depth.

**Comment by markelsanz14** [comment]
It seems like timedelta cannot be used with OmegaConf as a type annotation, but timedelta is what PTL uses (see link in PR description). Should I remove the type annotation? Or can I replace timedelta with something else?

**Comment by markelsanz14** [comment]
I modified the train_time_interval type annotation from Optional[timedelta] to Optional[str] after a discussion with @MaximumEntropy . Otherwise OmegaConf complains that timedelta is not a supported type annotation class. Please review again @nithinraok @titu1994

**Review by nithinraok — APPROVED** [review]
